src/Xamarin.Android.Build.Tasks/Utilities/ApplicationConfigNativeAssemblyGenerator.cs

```diff
@@ -39,7 +38,7 @@ protected override void WriteSymbols (StreamWriter output)
 			WriteData (output, AndroidPackageName, stringLabel);
 
 			WriteDataSection (output, "application_config");
-			WriteSymbol (output, "application_config", TargetProvider.GetStructureAlignment (true), fieldAlignBytes: 4, isGlobal: true, alwaysWriteSize: true, structureWriter: () => {
+			WriteSymbol (output, "application_config", TargetProvider.GetStructureAlignment (true), fieldAlignBytes: 8, isGlobal: true, alwaysWriteSize: true, structureWriter: () => {
```

Comment:
Changing `fieldAlignBytes` is something I'm not sure of here. It was crashing after I removed `uses_embedded_dsos`, and what was strange is the assembly generated had:
```assembly
.size	application_config, 20
```
It was crashing on this line in `monodroid-glue.cc` as if the structure was malformed:
```c
utils.monodroid_store_package_name (application_config.android_package_name);
```
Changing `fieldAlignBytes` to 8 adds padding that prevented my crash. But I don't know if this is padding too much. (or just wrong in a different way!)

Here is the latest assembly:
```assembly
	.arch	armv8-a
	.file	"environment.arm64-v8a.s"
	.section	.rodata..L.str.1,"aMS",@progbits,1
	.type	.L.str.1, @object
.L.str.1:
	.asciz	"com.xamarin.android.helloworld"
	.size	.L.str.1, 31
	.section	.data.application_config,"aw",@progbits
	.type	application_config, @object
	.p2align	3
	.global	application_config
application_config:
	/* uses_mono_llvm */
	.byte	0
	/* uses_mono_aot */
	.byte	0
	/* uses_assembly_preload */
	.byte	1
	/* is_a_bundled_app */
	.byte	0
	/* environment_variable_count */
	.zero	4
	.word	8
	/* system_property_count */
	.zero	4
	.word	0
	/* android_package_name */
	.zero	4
	.xword	.L.str.1
	.size	application_config, 32
	.section	.rodata..L.str.2,"aMS",@progbits,1
	.type	.L.str.2, @object
.L.str.2:
	.asciz	"0"
	.size	.L.str.2, 2
	.section	.data.mono_aot_mode_name,"aw",@progbits
	.global	mono_aot_mode_name
mono_aot_mode_name:
	.xword	.L.str.2
	.section	.rodata..L.str.3,"aMS",@progbits,1
	.type	.L.str.3, @object
.L.str.3:
	.asciz	"MONO_GC_PARAMS"
	.size	.L.str.3, 15
	.section	.rodata..L.str.4,"aMS",@progbits,1
	.type	.L.str.4, @object
.L.str.4:
	.asciz	"major=marksweep"
	.size	.L.str.4, 16
	.section	.rodata..L.str.5,"aMS",@progbits,1
	.type	.L.str.5, @object
.L.str.5:
	.asciz	"XAMARIN_BUILD_ID"
	.size	.L.str.5, 17
	.section	.rodata..L.str.6,"aMS",@progbits,1
	.type	.L.str.6, @object
.L.str.6:
	.asciz	"7a9af6a4-a48d-4dbd-8614-bae2ea47544a"
	.size	.L.str.6, 37
	.section	.rodata..L.str.7,"aMS",@progbits,1
	.type	.L.str.7, @object
.L.str.7:
	.asciz	"XA_HTTP_CLIENT_HANDLER_TYPE"
	.size	.L.str.7, 28
	.section	.rodata..L.str.8,"aMS",@progbits,1
	.type	.L.str.8, @object
.L.str.8:
	.asciz	"System.Net.Http.HttpClientHandler, System.Net.Http"
	.size	.L.str.8, 51
	.section	.rodata..L.str.9,"aMS",@progbits,1
	.type	.L.str.9, @object
.L.str.9:
	.asciz	"XA_TLS_PROVIDER"
	.size	.L.str.9, 16
	.section	.rodata..L.str.10,"aMS",@progbits,1
	.type	.L.str.10, @object
.L.str.10:
	.asciz	"btls"
	.size	.L.str.10, 5
	.section	.data.app_environment_variables,"aw",@progbits
	.type	app_environment_variables, @object
	.p2align	3
	.global	app_environment_variables
app_environment_variables:
	.xword	.L.str.3
	.xword	.L.str.4
	.xword	.L.str.5
	.xword	.L.str.6
	.xword	.L.str.7
	.xword	.L.str.8
	.xword	.L.str.9
	.xword	.L.str.10
	.size	app_environment_variables, 64
	.section	.data.app_system_properties,"aw",@progbits
	.type	app_system_properties, @object
	.p2align	3
	.global	app_system_properties
app_system_properties:
	.size	app_system_properties, 0

```

Reply:
The padding is correct

Reply:
Actually I think the env vars aren't coming through, due to the test failures I'm seeing:

![image](https://user-images.githubusercontent.com/840039/60919346-1637f380-a25b-11e9-9352-7d918dfb1820.png)

I will revisit the padding, and see if I can figure out what is happening.

Reply:
It appears that the structure needs to be packed more than it is, after removing the single byte. Here are the desired outputs for different architectures:

  * **arm64-v8a**
```assembly
application_config:
        .byte   0                       // 0x0
        .byte   0                       // 0x0
        .byte   0                       // 0x0
        .byte   0                       // 0x0
        .word   0                       // 0x0
        .word   0                       // 0x0
        .zero   4
        .xword  .L.str
        .size   application_config, 24
```

 * **armeabi-v7a**
```assembly
application_config:
        .byte   0                       @ 0x0
        .byte   0                       @ 0x0
        .byte   0                       @ 0x0
        .byte   0                       @ 0x0
        .long   0                       @ 0x0
        .long   0                       @ 0x0
        .long   .L.str
        .size   application_config, 16
```

 * **x86**
```assembly
application_config:
        .byte   0                       # 0x0
        .byte   0                       # 0x0
        .byte   0                       # 0x0
        .byte   0                       # 0x0
        .long   0                       # 0x0
        .long   0                       # 0x0
        .long   .L.str
        .size   application_config, 16
```

 * **x86-64**
```assembly
application_config:
        .byte   0                       # 0x0
        .byte   0                       # 0x0
        .byte   0                       # 0x0
        .byte   0                       # 0x0
        .long   0                       # 0x0
        .long   0                       # 0x0
        .zero   4
        .quad   .L.str
        .size   application_config, 24
```